workshop.wwwapi/Repository/Repository.cs

```diff
@@ -15,6 +15,12 @@ public async Task<IEnumerable<Patient>> GetPatients()
         {
             return await _databaseContext.Patients.ToListAsync();
         }
+
+        public async Task<Patient> GetPatientByID(int id)
+        {
+            IEnumerable<Patient> patients = await GetPatients();
+            return patients.FirstOrDefault(p => p.Id == id);
```

Comment:
this returns Patient or null, so update your function type to reflect this

Reply:
also your code here is better to call: `await _db.Patients.FindFirstOrDefault` rather than first actualyl retrieving ALL the patients and then filtering. Your approach is much more expensive computationally and time wise

Reply:
ok got it, thanks for feedback, I tried to update my code according to your comments